auction-app-frontend/src/components/CheckoutPaymentForm/CheckoutPaymentForm.jsx

```diff
@@ -0,0 +1,120 @@
+import { useState } from "react";
+import { useStripe, useElements, CardNumberElement, CardExpiryElement, CardCvcElement } from '@stripe/react-stripe-js';
+
+import { LoadingComponent, Button, ErrorComponent } from "src/components";
+import { BUTTON_VARIANTS, BUTTON_LABELS } from "src/constants";
+
+import "./style.scss";
+
+const CheckoutPaymentForm = ({ clientSecret, onPaymentSuccess }) =>
+{
+    const [errorMessage, setErrorMessage] = useState("");
+    const [successMessage, setSuccessMessage] = useState("");
+    const [cardNumberError, setCardNumberError] = useState("");
+    const [cardExpiryError, setCardExpiryError] = useState("");
+    const [cardCvcError, setCardCvcError] = useState("");
+    const [cardholderName, setCardholderName] = useState("");
+
+    const stripe = useStripe();
+    const elements = useElements();
+
+    const handleSubmit = async (event) =>
+    {
+        event.preventDefault();
+
+        if (!stripe || !elements) {
+            setErrorMessage("Stripe has not initialized");
+            return;
+        }
+
+        const cardElement = elements.getElement(CardNumberElement);
+
+        if (!cardElement) {
+            setErrorMessage("Card details not available");
+            return;
+        }
+
+        try {
+            const { token, error } = await stripe.createToken(cardElement, { name: cardholderName });
+            if (error) {
+                setErrorMessage(error.message);
+            } else {
+                setSuccessMessage("Payment confirmed successfully");
+                onPaymentSuccess(token);
+            }
+        } catch (error){
+            setErrorMessage(error.message || "An error occurred during payment confirmation.");
+        }
+    };
+
+    const handleCardNumberChange = (event) => {
+        setCardNumberError(event.error ? event.error.message : "");
+    };
+
+    const handleCardExpiryChange = (event) => {
+        setCardExpiryError(event.error ? event.error.message : "");
+    };
+
+    const handleCardCvcChange = (event) => {
+        setCardCvcError(event.error ? event.error.message : "");
+    };
+
+    return (
+        <div className="form-container">
+            <form onSubmit={ handleSubmit }>
+                <label className="body-regular">
+                    Name on Card
+                    <div className="input-container">
+                        <input
+                            type="text"
+                            value={ cardholderName }
+                            onChange={ (e) => setCardholderName(e.target.value) }
```

Comment:
Avoid declaring methods here
Create method for this case
If you do this
const handleInputChane = (e) => {
     setCardHolderName(e.target.value)
}

onChange = {handleInputChange)

This will work, and with re-render, this method won't be generated